Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -30,8 +30,96 @@ namespace AZ::Reflection
+        }
+        if (!serializeContext)
+        {
+            AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationRequests::GetSerializeContext);
+        }
+
+        const DocumentPropertyEditor::PropertyEditorSystemInterface* propertyEditorSystem =
+            AZ::Interface<DocumentPropertyEditor::PropertyEditorSystemInterface>::Get();
+
+        auto extractUuidProperty = [&propertyEditorSystem](auto& attributes, const AZ::Name& attributeName, AZ::Uuid& destination) -> bool
+        {
+            for (auto it = attributes.begin(); it != attributes.end(); ++it)
+            {
+                AZ::Name currentName = propertyEditorSystem->LookupNameFromId(it->first);
+                if (currentName == attributeName)
+                {
+                    AZ::AttributeReader reader(nullptr, it->second.get());
+                    if (reader.Read<AZ::Uuid>(destination))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        };
+
+        // Get our enum string value from the edit context if we're actually an enum
+        AZ::Uuid enumId;
+        extractUuidProperty(classElement->m_attributes, Name("EnumType"), enumId);
+
+        if (serializeContext && !enumId.IsNull())
+        {
+            // got the enumID, now get the enum's underlying type
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                const AZ::Edit::ElementData* enumElementData = editContext->GetEnumElementData(enumId);
+                if (enumElementData)
+                {
+                    AZ::Uuid underlyingTypeId;
+                    extractUuidProperty(classData->m_attributes, Name("EnumUnderlyingType"), underlyingTypeId);
+
+                    // Check all underlying enum storage types for the actual representation type to query
+                    if (!underlyingTypeId.IsNull() &&
+                        AZ::Utils::GetEnumStringRepresentation<AZ::u8, AZ::u16, AZ::u32, AZ::u64, AZ::s8, AZ::s16, AZ::s32, AZ::s64>(
+                            value, enumElementData, instance, underlyingTypeId))
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        // Just use our underlying AZStd::string if we're a string
+        if (classData->m_typeId == azrtti_typeid<AZStd::string>())
+        {
+            value = *reinterpret_cast<AZStd::string*>(instance);
+            return true;
+        }
+
+        // Fall back on using our serializer's DataToText
+        if (classElement)
+        {
+            if (auto& serializer = classData->m_serializer)
+            {
+                AZ::IO::MemoryStream memStream(instance, 0, classElement->m_dataSize);
+                AZStd::vector<char> buffer;
+                AZ::IO::ByteContainerStream<AZStd::vector<char>> outStream(&buffer);
```

Comment:
C++17 Class Template Argument Deduction avoids the need for specifying the template parameter.
```suggestion
                AZ::IO::ByteContainerStream outStream(&buffer);
```

Also you can use AZStd::string with the ByteContainerStream

Reply:
fixed, thanks!